### ISSUE TITLE
MariaDB stopping itself randomly

### ISSUE THREAD

**Issue by tech348712013870132** [bitnami/bitnami-docker-mariadb#210]
I have deployed version:
`docker.io/bitnami/mariadb:10.3.21-debian-10-r0`

The log of the running container is:

```
mariadb 15:22:15.84 
mariadb 15:22:15.85 Welcome to the Bitnami mariadb container
mariadb 15:22:15.85 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 15:22:15.86 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 15:22:15.86 Send us your feedback at containers@bitnami.com
mariadb 15:22:15.87 
mariadb 15:22:15.87 INFO  ==> ** Starting MariaDB setup **
mariadb 15:22:15.98 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 15:22:15.98 INFO  ==> Initializing mariadb database
mariadb 15:22:16.00 INFO  ==> Using persisted data
mariadb 15:22:16.02 INFO  ==> Running mysql_upgrade
mariadb 15:22:16.02 INFO  ==> Starting mariadb in background
mariadb 15:22:17.11 INFO  ==> Stopping mariadb
```

The container restarts indefinitely without any kind of error logs.

Does anyone know why the database is stopping itself without any kind of error logs?
If this is a expected behaviour how can the auto stopping be disabled?

**Comment by danjenkins** [comment]
I'm also seeing this using the ghost chart

**Comment by javsalgar** [comment]
Hi,

Could you try running it with the environment variable `BITNAMI_DEBUG=true`, this could give more information

**Comment by danjenkins** [comment]
For me it turned out to be a bad setup initially and my pvc wasn't being deleted due to statefulset so I just needed to delete the pvc and start over :)

**Comment by marcosbc** [comment]
That does make sense for me, if the first setup failed, second runs will fail for sure. @tech348712013870132 Is that your case as well?

**Comment by nelhousni** [comment]
> For me it turned out to be a bad setup initially and my pvc wasn't being deleted due to statefulset so I just needed to delete the pvc and start over :)

I got the same issue where mariadb stops as soon it's started, I delete all pvc then start over, it's working then, thanks.

**Comment by alexanderadam** [comment]
It didn't work for me either and I had the same output as in the first comment.
So I added `BITNAMI_DEBUG=true` to it and the true error became visible:

```
[…]
mysql.time_zone_transition                         OK
mysql.time_zone_transition_type                    OK
mysql.transaction_registry                         OK
Phase 2/7: Installing used storage engines... Skipped
Phase 3/7: Fixing views
mysql.user
Error    : The user specified as a definer ('root'@'localhost') does not exist
error    : Corrupt
Phase 4/7: Running 'mysql_fix_privilege_tables'

```

So I stopped using the `MARIADB_ROOT_USER` environment variable and everything went fine.

I used a new volume and therefore database, so I guess some script still relies on a `root` database user?

**Comment by marcosbc** [comment]
@alexanderadam Could you create a new Issue with more details? I.e. MariaDB version, docker-compose file where the issue happens, and whatever information can be useful for debugging this. Thanks!

**Comment by alexanderadam** [comment]
I'm not using `docker-compose` but I used `bitnami/mariadb:10.4.12` and I think I filled out

* `MARIADB_ROOT_USER`
* `MARIADB_ROOT_PASSWORD`
* `MARIADB_PASSWORD`
* `MARIADB_DATABASE`

as suggested in the README.
Apart from that I just had to run the image and wait until it tried to initialize the database.
So nothing special to do.

**Comment by marcosbc** [comment]
@alexanderadam Unfortunately we are unable to reproduce this issue. We ran the command below and it worked without issues:

```
docker run -ti -e MARIADB_ROOT_USER=myuser -e MARIADB_ROOT_PASSWORD=mypassword -e MARIADB_USER=myuser1 -e MARIADB_PASSWORD=mypassword1 -e MARIADB_DATABASE=mydb bitnami/mariadb:latest
```

Please create a ticket if you continue finding this issue, with the detailed steps/commands to reproduce it if possible.

**Comment by alexanderadam** [comment]
Hey @marcosbc,

yes, it's probably because you didn't use [the image I mentioned](https://github.com/bitnami/bitnami-docker-mariadb/issues/210#issuecomment-612106809). :wink: 
Try this instead:

```bash
$ docker run -ti -e MARIADB_ROOT_USER=myuser \
                 -e MARIADB_ROOT_PASSWORD=mypassword \
                 -e MARIADB_USER=myuser1 \
                 -e MARIADB_PASSWORD=mypassword1 \
                 -e MARIADB_DATABASE=mydb \
                 -e BITNAMI_DEBUG=true \
                 bitnami/mariadb:10.4.12
```

**Comment by marcosbc** [comment]
You're right @alexanderadam. We were finally able to reproduce it and are working on a fix. We'll keep you posted.

**Comment by marcosbc** [comment]
@alexanderadam Check out the MariaDB 10.4 container image that was just released with the fix:

* 10.4.12-debian-10-r75

Thanks again for the bug report!

**Comment by yelhouti** [comment]
@marcosbc please update: https://github.com/bitnami/charts/blob/master/bitnami/mariadb/Chart.yaml to use the latest versions as many charts depend on it, and it keeps failing

**Comment by marcosbc** [comment]
@yelhouti Could you clarify what you mean with "many charts depend on it, and it keeps failing"?

Note that this issue is specifically affecting 10.4, and the current MariaDB chart uses 10.3.x so the issue should not be present. And if you specify a different tag (i.e. 10.4.12-debian-10-r75), then it should also work without issues due to the fix we included.

Let us know if there's anything we're missing.

**Comment by tanyadream** [comment]
> For me it turned out to be a bad setup initially and my pvc wasn't being deleted due to statefulset so I just needed to delete the pvc and start over :)

how do you delete and restart pvc or all the helm release?

I have the same:
helm install dev-release bitnami/ghost --set ghostHost=127.0.0.1,mariadb.master.persistence.size=100Mi,persistence.size=100Mi,global.storageClass=local-storage,ghostPassword=a1234567890,mariadb.db.password=a1234567890,smtpPassword=a1234567890,mariadb.rootUser.password=a1234567890,ghostUsername=user@example.com,ghostPort=2368,smtpFromAddress=blog@whatismyip.li,smtpHost=smtp.zoho.eu,smtpPort=465,smtpUser=blog@whatismyip.li,smtpPassword=ay9SQQbMrzPn8nur,smtpService=gmail,readinessProbe.initialDelaySeconds=150,livenessProbe.initialDelaySeconds=150 -n ghost

mariadb 11:27:59.64
mariadb 11:27:59.64 Welcome to the Bitnami mariadb container
mariadb 11:27:59.64 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 11:27:59.68 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 11:27:59.68
mariadb 11:27:59.70 INFO ==> ** Starting MariaDB setup **
mariadb 11:27:59.76 INFO ==> Validating settings in MYSQL_/MARIADB_ env vars
mariadb 11:27:59.77 INFO ==> Initializing mariadb database
mariadb 11:27:59.78 WARN ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 11:27:59.79 INFO ==> Using persisted data
mariadb 11:27:59.82 INFO ==> Running mysql_upgrade
mariadb 11:27:59.83 INFO ==> Starting mariadb in background
mariadb 11:28:01.92 INFO ==> Stopping mariadb

**Comment by carrodher** [comment]
PVs/PVCs are not deleted when `helm delete` is executed, you need to manually remove them by 1) listing the existing ones with `kubectl get pvc` and 2) removing the desired ones by running `kubectl delete pvc NAME1 NAME2 NAME3`. The same approach can be applied to PVs.

**Comment by alexanderadam** [comment]
By the way the second point

>     2. It should be visible _without_ having to set `BITNAMI_DEBUG` if something like this happens.

is still relevant and unfixed (at least in `10.4` tags).
Errors won't be logged unless `BITNAMI_DEBUG` is set.

I just started another container and everything that's not starting with `INFO` is not shown without `BITNAMI_DEBUG`:

![mariadb 09:26:44.78 INFO  ==> Running mysql_upgrade mariadb 09:26:44.78 INFO  ==> Starting mariadb in background 2021-05-21  9:26:44 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.4.19-MariaDB) starting as process 49 ... 2021-05-21  9:26:44 0 [Note] InnoDB: Using Linux native AIO 2021-05-21  9:26:44 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins 2021-05-21  9:26:44 0 [Note] InnoDB: Uses event mutexes 2021-05-21  9:26:44 0 [Note] InnoDB: Compressed tables use zlib 1.2.11 2021-05-21  9:26:44 0 [Note] InnoDB: Number of pools: 1 2021-05-21  9:26:44 0 [Note] InnoDB: Using SSE2 crc32 instructions 2021-05-21  9:26:44 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts) 2021-05-21  9:26:44 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M 2021-05-21  9:26:44 0 [Note] InnoDB: Completed initialization of buffer pool 2021-05-21  9:26:44 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority(). 2021-05-21  9:26:44 0 [Note] InnoDB: 128 out of 128 rollback segments are active. 2021-05-21  9:26:44 0 [Note] InnoDB: Creating shared tablespace for temporary tables 2021-05-21  9:26:44 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ... 2021-05-21  9:26:44 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB. 2021-05-21  9:26:44 0 [Note] InnoDB: 10.4.19 started; log sequence number 47075; transaction id 8 2021-05-21  9:26:44 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool 2021-05-21  9:26:44 0 [Note] InnoDB: Buffer pool(s) load completed at 210521  9:26:44 2021-05-21  9:26:44 0 [Note] Plugin 'FEEDBACK' is disabled. 2021-05-21  9:26:44 0 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded 2021-05-21  9:26:44 0 [ERROR] Can't open and lock privilege tables: Table 'mysql.servers' doesn't exist 2021-05-21  9:26:44 0 [Note] Server socket created on IP: '127.0.0.1'. 2021-05-21  9:26:44 0 [ERROR] Fatal error: Can't open and lock privilege tables: Table 'mysql.db' doesn't exist 2021-05-21  9:26:44 0 [ERROR] Aborting Version check failed. Got the following error when calling the 'mysql' command line client ERROR 2013 (HY000): Lost connection to MySQL server at 'handshake: reading initial communication packet', system error: 104 FATAL ERROR: Upgrade failed
](https://user-images.githubusercontent.com/372620/119115897-fb726080-ba27-11eb-81cf-d997a6fd081d.png)

So the user usually sees that the container starts and then it suddenly stops without mentioning a reason.

**Comment by carrodher** [comment]
Hi, this is the desired approach for logging, it is not an issue that should be solved.
When MariaDB is deployed we can differentiate two different steps.

The first one is the MariaDB **initialization**:
```
mariadb_1  | mariadb 06:44:37.99
mariadb_1  | mariadb 06:44:38.00 Welcome to the Bitnami mariadb container
mariadb_1  | mariadb 06:44:38.00 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb_1  | mariadb 06:44:38.00 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb_1  | mariadb 06:44:38.00
mariadb_1  | mariadb 06:44:38.00 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 06:44:38.01 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 06:44:38.02 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
mariadb_1  | mariadb 06:44:38.02 INFO  ==> Initializing mariadb database
mariadb_1  | mariadb 06:44:38.03 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1  | mariadb 06:44:38.03 INFO  ==> Installing database
mariadb_1  | mariadb 06:44:40.58 INFO  ==> Starting mariadb in background
mariadb_1  | mariadb 06:44:42.60 INFO  ==> Configuring authentication
mariadb_1  | mariadb 06:44:42.64 INFO  ==> Running mysql_upgrade
mariadb_1  | mariadb 06:44:44.38 INFO  ==> Stopping mariadb
mariadb_1  | mariadb 06:44:49.55 INFO  ==> ** MariaDB setup finished! **
mariadb_1  | mariadb 06:44:49.57 INFO  ==> ** Starting MariaDB **
```

Then MariaDB (as a service itself) is launched:
```
mariadb_1  | mariadb 06:44:49.57 INFO  ==> ** Starting MariaDB **
mariadb_1  | 2021-05-24  6:44:49 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.4.19-MariaDB) starting as process 1 ...
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Using Linux native AIO
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Uses event mutexes
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Number of pools: 1
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Using SSE2 crc32 instructions
mariadb_1  | 2021-05-24  6:44:49 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Waiting for purge to start
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: 10.4.19 started; log sequence number 60976; transaction id 20
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2021-05-24  6:44:49 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1  | 2021-05-24  6:44:49 0 [Note] InnoDB: Buffer pool(s) load completed at 210524  6:44:49
mariadb_1  | 2021-05-24  6:44:49 0 [Note] Server socket created on IP: '0.0.0.0'.
mariadb_1  | 2021-05-24  6:44:49 0 [Warning] 'proxies_priv' entry '@% root@9c604049dbba' ignored in --skip-name-resolve mode.
mariadb_1  | 2021-05-24  6:44:49 0 [Note] Reading of all Master_info entries succeeded
mariadb_1  | 2021-05-24  6:44:49 0 [Note] Added new Master_info '' to hash table
mariadb_1  | 2021-05-24  6:44:49 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1  | Version: '10.4.19-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
```

During the initialization, there are several scripts executed (you can find them at https://github.com/bitnami/bitnami-docker-mariadb/tree/master/10.4/debian-10/rootfs/opt/bitnami/scripts). In this step, only the messages with log level set to `info` (or higher are shown). If you want to see other logs, it is needed to set the `BITNAMI_DEBUG` environment variable. During this step, some actions are performed in order to configure the database so increasing the verbosity in this step is not very useful unless there are some issues, in those cases, it is recommended to set the `BITNAMI_DEBUG` environment variable, but just in case some issue appears.

Once the initialization is done, we are just sending to `stdout` the MariaDB logs, at that point there are not Bitnami specific scripts executed so once everything is deployed, all the information provided by the MariaDB service should be shown.

**Comment by alexanderadam** [comment]
So it's absolutely not possible to log stderr of MariaDB? :thinking:

**Comment by carrodher** [comment]
During the initialization step no, once the container is deployed all the MariaDB logs are redirected to stdout and are shown in the console.

**Comment by grooverdan** [comment]
Note: Upstream fixed original issue in [10.4.13](https://mariadb.com/kb/en/mysqluser-table) and in #257 I've requested the removal of the work around.

**Comment by 77557231** [comment]
> @alexanderadam Check out the MariaDB 10.4 container image that was just released with the fix:
> 
> * 10.4.12-debian-10-r75
> 
> Thanks again for the bug report!

I have deployed version:
docker.io/bitnami/mariadb:10.5.11-debian-10-r25
helm chart version: 9.3.19

have the same problem, but user is root.
The log of the running pod with debug is:
root@B2U1200900577:~# kubectl -n mcd-infra logs mariadb-0 mariadb
mariadb 03:09:15.15 
mariadb 03:09:15.16 Welcome to the Bitnami mariadb container
mariadb 03:09:15.16 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 03:09:15.16 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 03:09:15.17 
mariadb 03:09:15.17 INFO  ==> ** Starting MariaDB setup **
mariadb 03:09:15.20 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 03:09:15.22 INFO  ==> Initializing mariadb database
mariadb 03:09:15.22 DEBUG ==> Ensuring expected directories/files exist
mariadb 03:09:15.25 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 03:09:15.25 INFO  ==> Using persisted data
mariadb 03:09:15.29 INFO  ==> Running mysql_upgrade
mariadb 03:09:15.30 INFO  ==> Starting mariadb in background
2021-12-08  3:09:15 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.5.11-MariaDB-log) starting as process 47 ...
2021-12-08  3:09:15 0 [Note] InnoDB: Uses event mutexes
2021-12-08  3:09:15 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-12-08  3:09:15 0 [Note] InnoDB: Number of pools: 1
2021-12-08  3:09:15 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2021-12-08  3:09:15 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2021-12-08  3:09:15 0 [Note] InnoDB: Using Linux native AIO
2021-12-08  3:09:15 0 [Note] InnoDB: Initializing buffer pool, total size = 37580963840, chunk size = 134217728
2021-12-08  3:09:16 0 [Note] InnoDB: Completed initialization of buffer pool
2021-12-08  3:09:16 0 [Note] InnoDB: 128 rollback segments are active.
2021-12-08  3:09:16 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-12-08  3:09:16 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-12-08  3:09:16 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-12-08  3:09:16 0 [Note] InnoDB: 10.5.11 started; log sequence number 12265155; transaction id 11348
2021-12-08  3:09:16 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-12-08  3:09:16 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2021-12-08  3:09:16 0 [Note] Server socket created on IP: '127.0.0.1'.
2021-12-08  3:09:16 0 [Warning] 'proxies_priv' entry '@% root@210235a2cs621bf00147' ignored in --skip-name-resolve mode.
2021-12-08  3:09:16 0 [Note] InnoDB: Buffer pool(s) load completed at 211208  3:09:16
2021-12-08  3:09:16 0 [Note] Reading of all Master_info entries succeeded
2021-12-08  3:09:16 0 [Note] Added new Master_info '' to hash table
2021-12-08  3:09:16 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
Version: '10.5.11-MariaDB-log'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
2021-12-08  3:09:17 3 [Warning] Access denied for user 'root'@'localhost' (using password: YES)
Version check failed. Got the following error when calling the 'mysql' command line client
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
FATAL ERROR: Upgrade failed
mariadb 03:09:17.36 INFO  ==> Stopping mariadb
2021-12-08  3:09:17 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
2021-12-08  3:09:17 0 [Note] Event Scheduler: Purging the queue. 0 events
2021-12-08  3:09:17 0 [Note] InnoDB: FTS optimize thread exiting.
2021-12-08  3:09:17 0 [Note] InnoDB: Starting shutdown...
2021-12-08  3:09:17 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
2021-12-08  3:09:17 0 [Note] InnoDB: Buffer pool(s) dump completed at 211208  3:09:17
2021-12-08  3:09:18 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2021-12-08  3:09:18 0 [Note] InnoDB: Shutdown completed; log sequence number 12265167; transaction id 11349
2021-12-08  3:09:18 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete
